pkg/appStore/deployment/AppStoreDeploymentService.go

```diff
@@ -615,3 +619,53 @@ func (impl AppStoreDeploymentServiceImpl) createEnvironmentIfNotExists(installAp
 
 	return envCreateRes.Id, nil
 }
+
+func (impl AppStoreDeploymentServiceImpl) RollbackApplication(ctx context.Context, request *openapi2.RollbackReleaseRequest,
+	installedApp *appStoreBean.InstallAppVersionDTO, userId int32) (bool, error) {
+
+	dbConnection := impl.installedAppRepository.GetConnection()
+	tx, err := dbConnection.Begin()
+	if err != nil {
+		return false, err
+	}
+	// Rollback tx on error.
+	defer tx.Rollback()
+
+	// Rollback starts
+	installedAppVersion, err := impl.installedAppRepository.GetInstalledAppVersion(int(request.GetInstalledAppVersionId()))
+	if err != nil {
+		impl.logger.Errorw("error while fetching chart installed version", "error", err)
+		return false, err
+	}
+
+	var valuesYaml string
+	var success bool
+	if installedApp.AppOfferingMode == util2.SERVER_MODE_HYPERION {
+		valuesYaml, success, err = impl.appStoreDeploymentHelmService.RollbackRelease(ctx, installedApp, request.GetVersion())
+		if err != nil {
+			impl.logger.Errorw("error while rollback helm release", "error", err)
+			return false, err
+		}
+	} else {
+		// TODO : handle acd
```

Comment:
???

Reply:
Devtron chart's app rollback feature (Just like hyperion helm app) - to be code by Vikram.